src/jit/hwintrinsic.hpp

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/*****************************************************************************
+*
+*  Convert the type returned from the VM to a var_type using one to one mapping
+*  without coalescing unsigned integral types into signed int or long.
+*/
+
+inline var_types JITtype2varTypeExact(CorInfoType type)
```

Comment:
Strange choice of file for this function. The function name is rather generic (what does "exact" exactly means anyway?) but it's placed in a "hwintrinsic" file.

It would be better for the name to mention "SIMD" to avoid accidental use of this function in contexts that do not have anything to do with SIMD. Then you could also restrict the mapping to types that make sense as SIMD base types and return `TYP_UNDEF` for any other types (e.g. for `TYP_REF`).

Reply:
Common code for HW intrinsic implementation is located in `hwintrinsic.cpp` file. This code is also common for all architectures and since function should be inlined I have followed jit convention of placing it into `.hpp` file.

Naming is an open question and I do not have any strong preferences for it.